mypy/docstrings.py

```diff
@@ -0,0 +1,27 @@
+from typing import List, Tuple, Dict
+from mypy.types import Type, CallableType, AnyType
+from mypy.nodes import Argument
+
+
+def parse_docstring(docstring: str, line: int) -> Tuple[Dict[str, Type], Type]:
+    """
+    Parse a docstring and return type representations.  This function can
+    be overridden by third-party tools which aim to add typing via docstrings.
+
+    Returns a 2-tuple: dictionary of arg name to Type, and return Type.
+    """
+    return None, None
+
+
+def make_callable(args: List[Argument], type_map: Dict[str, Type],
+                  ret_type: Type) -> CallableType:
+    if type_map is not None:
```

Comment:
I see, it's because you're emulating the result of parse_type_comment() which in turn calls parse_signature() which returns a Callable.

So here's another exhortation to return strings and let mypy parse the strings. Strings are what you have in your docstring and strings provide a much more stable API between an app (e.g. mypy) and an extension (your code for parsing docstrings) than Type objects. If we change the signature of Callable or we change the meaning of Type, your extension will break, and we can't make any promises that such internals won't change (not until we've had much more experience with extensions). But strings are pretty stable and you can always construct strings. So I'm saying you should just return a string that's got the same format as a type comment, i.e. `# type: (blah, blah) -> blah` (but without the `# type:` prefix).